src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/CollectF1R2Counts.java

```diff
@@ -0,0 +1,233 @@
+            return;
+        }
+
+        // If we got here, we have an alt site with a single alt base
+        final Nucleotide altBase = Nucleotide.valueOf(BaseUtils.baseIndexToSimpleBase(altBaseIndex));
+
+        final int refCount = baseCounts[refBase.ordinal()];
+        final int altCount = baseCounts[altBaseIndex];
+        Utils.validate(altCount > 0, "We must have a nonzero alt read but got " + altCount);
+
+        final int refF1R2 = pileup.getNumberOfElements(pe -> Nucleotide.valueOf(pe.getBase()) == refBase && ReadUtils.isF1R2(pe.getRead()));
+        final int altF1R2 = pileup.getNumberOfElements(pe -> Nucleotide.valueOf(pe.getBase()) == altBase && ReadUtils.isF1R2(pe.getRead()));
+
+        // If there's only one alt read at depth > minimumDepthForOptimization, we store the data differently to save space
+        if (altCount == 1) {
+            final ReadOrientation type = altF1R2 == 1 ? F1R2 : F2R1;
+            depthOneAltHistograms.get(refContext).get(new Pair<>(altBase, type))
+                    .increment(depth <= F1R2FilterConstants.maxDepth ? depth : F1R2FilterConstants.maxDepth);
+            return;
+        }
+
+        try {
+            altTableWriter.writeRecord(new AltSiteRecord(refContext, refCount, altCount, refF1R2, altF1R2, altBase));
+        } catch (IOException e) {
+            throw new UserException("Encountered an IO Exception writing to the alt data table", e);
+        }
+    }
+
+    @Override
+    public Object onTraversalSuccess() {
+        refSiteHistograms.values().forEach(h -> refMetricsFile.addHistogram(h));
+        refMetricsFile.write(refMetricsOutput);
+
+        depthOneAltHistograms.values().forEach(hs -> hs.values().forEach(h -> altMetricsFile.addHistogram(h)));
+        altMetricsFile.write(altMetricsOutput);
+
+        return "SUCCESS";
+    }
+
+    @Override
+    public void closeTool() {
+        if (altTableWriter != null) {
+            try {
+                altTableWriter.close();
+            } catch (IOException e) {
+                throw new UserException("Encountered an IO exception while closing the alt table writer", e);
+            }
+        }
+    }
+
+    /**
+     * Use a series of heuristics to detect a bad pileup.
+     */
+    private boolean isPileupGood(final ReadPileup pileup, final ReferenceContext referenceContext){
+        // This case should not happen, as AlignmentContext should come filtered, but it does happen once in a while
+
+        final int[] baseCounts = pileup.getBaseCounts();
+        final int depth = (int) MathUtils.sum(baseCounts);
+
+        // If observe any bases that are neither ref nor alt, the site is no good so filter
+        final long numAlleles = Arrays.stream(baseCounts).filter(i -> i > 0).count();
+
+        List<Integer> mappingQualities = Ints.asList(pileup.getMappingQuals());
+
+        boolean isIndel = pileup.getNumberOfElements(pe -> pe.isDeletion() || pe.isAfterInsertion() || pe.isBeforeDeletionStart()) > 0;
```

Comment:
This also breaks down at very high depth.

Reply:
This one I want to fix as follows:

```java
        // If more than 1% of the reads is indel then consider this site an indel
        final int indelThreshold = depth/100;
        boolean isIndel = pileup.getNumberOfElements(pe -> pe.isDeletion() || pe.isAfterInsertion() || pe.isBeforeDeletionStart()) > indelThreshold;
```

Reply:
Sounds good.